aten/src/ATen/native/native_functions.yaml

```diff
@@ -10541,6 +10541,21 @@
   python_module: nn
   variants: function
 
+- func: pad(Tensor self, Tensor pad_width, str mode='constant', *, Tensor? constant_values=None) -> Tensor
+  variants: function, method
+  dispatch:
+    CPU, CUDA: pad
+  device_check: NoCheck  # pad_width can only be on CPU, for better performance
+
+- func: pad.out(Tensor self, Tensor pad_width, str mode='constant', *, Tensor? constant_values=None, Tensor(a!) out) -> Tensor(a!)
+  dispatch:
+    CPU, CUDA: pad_out
+  device_check: NoCheck  # pad_width can only be on CPU, for better performance
+
+- func: _expand_pad_specifier(Tensor pad_specifier, str arg_name, int ndim) -> Tensor
```

Comment:
Why specify this in native_functions.yaml?

Reply:
Before, when I was calling `at::native::_expand_pad_specifier` directly in `pad_backward`, @albanD said that I shouldn't call into `at::native` from a backward function. I figured that putting it in native_functions.yaml and calling it with `at::_expand_pad_specifier` would solve that. Is there a better way?

Reply:
I see. @albanD, is our recommendation to put all functions called from backward in native_functions.yaml or do we have a better place to expose them?

If this op is no longer being called from backward, however, I think we can remove it here either way

Reply:
> If this op is no longer being called from backward, however, I think we can remove it here either way

It is still being called from the backward

Reply:
Unless your backward function has a formula defined for it, it needs to be a composite op. Which means that you need to only use Tensor and differentiable ops.
at::native function are not differentiable ops (as they don't go through the dispatcher).
Hence my recommendation.

Reply:
Looking at pad_backward, it's ok to call non-dispatched _expand_pad_specifier there (and it's still calling `at::native::pad_width_to_inner_slices`). Both those functions are only used to compute `slices` for `grad.index(slices)`, so `slices` will be properly recorded for subsequent higher order backward calls.

Reply:
Done